src/main/java/javaslang/collection/List.java

```diff
@@ -1348,6 +1354,32 @@ public T head() {
         }
 
         @Override
+        public Tuple2<List<T>, List<T>> splitAt(Predicate<? super T> predicate) {
+            Tuple2<List<T>, List<T>> t = splitByPredicateFirstRound(this, predicate);
+            return Tuple.of(t._1.reverse(), t._2);
+        }
+
+        @Override
+        public Tuple2<List<T>, List<T>> splitAtInclusive(Predicate<? super T> predicate) {
+            Tuple2<List<T>, List<T>> t = splitByPredicateFirstRound(this, predicate);
+            if (!t._2.isEmpty()) {
+                return Tuple.of(t._1.prepend(t._2.head()).reverse(), t._2.tail());
+            } else {
+                return Tuple.of(t._1.reverse(), t._2);
+            }
+        }
+
+        private static <T> Tuple2<List<T>, List<T>> splitByPredicateFirstRound(List<T> source, Predicate<? super T> predicate) {
```

Comment:
I would name this `splitByPredicateReversed` to give a hint that the first component is split